README.md

```diff
@@ -1,4 +1,371 @@
+```
+3. Autorest  
+it can be installed by running `npm install -g autorest@latest` 
+4. Repos (Optional)  
+if users only want to try out the generated CLI extensions, they can also build the wheel artifact by themselves with `python setup.py sdist bdist_wheel` in the azure-cli-extensions/src/{serviceName} folder.  
+* Users need to prepare their swagger and prepare  
+[azure-cli-extensions](https://github.com/Azure/azure-cli-extensions) repo if targeting at Azure CLI extensions development. Or the generated code needs some manual customization after trying out.   
+[azure-cli](https://github.com/Azure/azure-cli) repo if targeting at Azure CLI main repo modules development.   
+
+
+## Authoring readme files
+Because the Autorest.Az depends on Autorest.Clicommon and Autorest.Python, it needs three readme files the readme.az.md, readme.cli.md and readme.python.md to be ready before we use it.
+
+Users can refer to [this document](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/code-gen/configure-cli.md) for more details. 
+
+## Generating the code
+1. A simple command for authoring Azure CLI extensions would be  
+`autorest --az --use=@autorest/az@latest <path-to-the-swagger-readme.md> --azure-cli-extension-folder=<path-to-the-azure-cli-extension-repo>`
+
+2. A simple command for authoring Azure CLI main modules would be  
+`autorest --az --use=@autorest/az@latest <path-to-the-swagger-readme.md> --azure-cli-folder=<path-to-the-azure-cli-repo>`
+
+The `--use=@autorest/az@latest` will use the latest `@autorest/az` public releases, private releases can be found in [github releases](https://github.com/Azure/autorest.az/releases) if users want to try out some private releases for preview some new features and fixes.  
+
+Other command line options can be found in the [generate with different options doc](https://github.com/Azure/autorest.az/blob/master/doc/how-to-generate-with-different-options.md)
+
+
+## Building the code
+If it's try out mode, 
+``` 
+cd <az-output-folder> // the az-output-folder you have specified in your readme.az.md 
+python setup.py sdist bdist_wheel // you will find a wheel file in your local dist folder
+az extension add <path-to-the-wheel-file>
+``` 
+If it's for azure-cli-extensions development, you need  
+```
+azdev setup -r ./azure-cli-extensions -c ./azure-cli 
+// where the ./azure-cli-extensions is the path to azure-cli-extensions repo and ./azure-cli is the path to azure-cli repo, you don't need to specify them both if you are not working on both of them
+azdev extension add <extension-name> // for Azure CLI extensions
+// extension add step is needed for developing Azure CLI main repo modules 
+```
+## Play with Azure CLI
+Now you can run `az <extension-name> -h` to see all the commands and parameters. You can also find a report.md in generated azext_{extensionName} folder, which contains an overview of all the generated command groups, commands and parameters.   
```

Comment:
We may have the scenario that the extension name is different from main command group name?

Reply:
Yes. I changed the doc. Just thought if it may be confusion for users if we say main command group name. In the Folder Customization, I mentioned each place holder can be changable.